fabric_plugin/tests/tests.py

```diff
@@ -507,6 +510,43 @@ def test_run_script_download_resource(self):
             })
         self.assertEqual(return_value, 'content')
 
+    def test_run_script_inputs_as_env_variables(self):
+        def test(value):
+            return_value, _ = self._execute(
+                'test.run_script',
+                script_path='scripts/script.sh',
+                custom_input=value,
+                process={
+                    'env': {
+                        'test_operation': self._testMethodName
+                    }
+                })
+            self.assertEqual(return_value if isinstance(value, basestring)
+                             else json.loads(return_value), value)
+        test('string-value')
+        test([1, 2, 3])
+        test({'key': 'value'})
+
+    def test_run_script_inputs_as_env_variables_process_env_override(self):
+        test_operation = self.test_run_script_inputs_as_env_variables.__name__
```

Comment:
I think having a duplicate function in script_test.sh is better than the logic applied here which deviates from all the rest of the tests.